src/python/pants/backend/python/goals/package_pex_binary_integration_test.py

```diff
@@ -456,3 +457,47 @@ def test_extra_build_args(rule_runner: PythonRuleRunner) -> None:
 
     assert additional_args[-2] == "--example-extra-arg"
     assert additional_args[-1] == "value-goes-here"
+
+
+def test_package_with_python_provider() -> None:
+    # Per https://github.com/pantsbuild/pants/issues/21048, test packaging a local/unconstrained pex
+    # binary when using a Python that isn't automatically visible on $PATH (using the PBS provider
+    # as just one way to get such a Python)
+
+    rule_runner = PythonRuleRunner(
+        rules=[
+            *package_pex_binary.rules(),
+            *pex_from_targets.rules(),
+            *target_types_rules.rules(),
+            *core_target_types_rules(),
+            *pbs.rules(),
```

Comment:
This test uses its own rule runner rather than the `rule_runner` fixture to be able to install these PBS rules, rather than needing to include them for the rest of the tests in this file.